skyportal/handlers/api/spectrum.py

```diff
@@ -176,6 +177,100 @@ def delete(self, spectrum_id):
         return self.success()
 
 
+class ASCIIHandler(object):
+    def spec_from_ascii_request(self):
+        """Helper method to read in Spectrum objects from ASCII POST."""
+        json = self.get_json()
+
+        try:
+            json = SpectrumAsciiFilePostJSON.load(json)
+        except ValidationError as e:
+            raise ValidationError(
+                'Invalid/missing parameters: ' f'{e.normalized_messages()}'
+            )
+
+        obj = Source.get_obj_if_owned_by(json['obj_id'], self.current_user)
+        if obj is None:
+            raise ValidationError('Invalid Obj id.')
+
+        instrument = Instrument.query.get(json['instrument_id'])
+        if instrument is None:
+            raise ValidationError('Invalid instrument id.')
+
+        group_ids = json.pop('group_ids')
+        user_group_ids = [g.id for g in self.current_user.accessible_groups]
+        for group_id in group_ids:
+            if group_id not in user_group_ids:
+                raise PermissionError('Insufficient permissions.')
+
+        filename = json.pop('filename')
+        ascii = json.pop('ascii')
+        spec = Spectrum.from_ascii(filename, data=ascii, **json)
+        return spec
+
+
+class SpectrumASCIIFileHandler(BaseHandler, ASCIIHandler):
+    @permissions(['Upload data'])
+    def post(self):
+        """
+        ---
+        description: Upload spectrum from ASCII file
```

Comment:
Do you want the upload to fail even if it cannot get parsed? Perhaps a non-standard file is getting uploaded. Even if it cannot be displayed a user may want to store it.

Reply:
Delegated this to a future PR.